### ISSUE TITLE
Custom styling of docs to look more like Wagtail

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#4081]
This is just an idea, would be nice to make the docs look a more like the Wagtail admin theming (or even the main Wagtail website style).

I have got a proof of concept working locally...

**Current**
![wagtail docs - current style](https://user-images.githubusercontent.com/1396140/33416940-1fc3754a-d5da-11e7-862d-9c502782821b.png)

**With basic styles**
![wagtail docs - custom style](https://user-images.githubusercontent.com/1396140/33416946-27210c08-d5da-11e7-9733-d176b9175e33.png)

## Implementation
Sphinx makes this reasonably easy to do this with their setup api documented [here](http://www.sphinx-doc.org/en/stable/extdev/appapi.html#sphinx.application.Sphinx.add_stylesheet).

docs/_static/css/custom.css - create new css file, just basic styling for example
```
html  {
  background-color: #e6e6e6;
}
body.wy-body-for-nav {
  -webkit-font-smoothing: antialiased; /* Do not remove! */
  font-family: Open Sans, Arial, sans-serif;
  line-height: 1.5em;
}
div.wy-nav-side {
  background-color: #333;  /*color-grey-1*/
}
div.wy-side-nav-search {
  background-color: #43b1b0; /*color-teal*/
}
```

docs/conf.py - declare additional stylesheet
```
html_static_path = ['_static']
#...
def setup(app):
    app.add_stylesheet('css/custom.css')
```

**Comment by jjanssen** [comment]
This looks great! Do not know for the rest of the team but I'm happy to go forward with this.

**Comment by BertrandBordage** [comment]
Just an improvement suggestion: the version number below the wagtail is not readable.

**Comment by lb-** [comment]
These are the correct screenshots that reflect the CSS used in the PR, Version number is much easier to read here.


![wagtail documentation - mobile view](https://user-images.githubusercontent.com/1396140/34027405-9b3c98be-e1b0-11e7-8a80-43c26803bf11.png)

![wagtail documentation - desktop view](https://user-images.githubusercontent.com/1396140/34027406-9b6be5ce-e1b0-11e7-94db-54ed73bce870.png)

**Comment by lb-** [comment]
Merged #4094
